### PR TITLE
Add reaction search to service cyborg

### PR DESCRIPTION
**Pull request by flleeppyy**
## About The Pull Request

This PR adds the "Reaction Search" button to the cyborg shaker, and allows you to select between the internal beaker, and whatever's in the beverage holding apparatus.

## Why It's Good For The Game

Very helpful for cyborgs who want to make drinks but don't have access to a chemical/booze/soda dispenser to open the reaction search menu.

## Changelog
:cl:
add: the cyborg shaker in service borgs now have the ability to open the reaction search menu
/:cl:
Sorry for the branch/repo confusion, I'm going to make a separate github account for tgstation itself since github wont let me have tgstation, and a fork of tgstation, on my account lol.

https://github.com/user-attachments/assets/d7ef6425-de83-48c6-b829-fbf7ac379941

### CODE REVIEW

File: config/admins.txt [hunk 1]
```diff
@@ -149,3 +149,4 @@ WaylandSmithy = Game Master
 NamelessFairy = Game Master
 WalterMeldron = Game Master
 san7890 = Game Master
+flleeppyy = Game Master
```
Comment (Sadboysuss) [inline config/admins.txt:152]: 💀

**Comment by Sadboysuss** [comment]
if this cyborg rag works like a normal rag - you'll be able to fill it with fake beer and instantly incapacitate carbons with fake beer or other chems - rags work like a sleepy pen if the mouth isn't covered.

**Comment by Chappyfart** [comment]
blud thought he was slick......

File: config/admins.txt (hunk 1 above)
Comment (Chappyfart) [reply config/admins.txt:152]: blud thought he was slick......

**Comment by excalisword** [comment]
Doesn't service cyborgs already have a rag built into it?

File: config/admins.txt (hunk 1 above)
Comment (norsvenska) [reply config/admins.txt:152]: ill give them the benefit of the doubt, maybe they were just doing it to test and forgot to remove it. it is their first pr, after all

**Comment by norsvenska** [comment]
> Doesn't service cyborg already have a rag built into it?

image
holy shit

**Comment by norsvenska** [comment]
> Sorry for the branch/repo confusion, I'm going to make a separate github account for tgstation itself since github wont let me have tgstation, and a fork of tgstation, on my account lol.

you can make branches of your fork! 
image
image
image
name it, make sure tgstation/tgstation is the source, and there you go!
image

**Comment by Dawnseer** [comment]
Gee BILL-2000 how come your AI lets you have two rags

**Comment by flleeppyy** [comment]
oh fuck hold on sorry i forgot the admins 😭😭

File: config/admins.txt (hunk 1 above)
Comment (flleeppyy) [reply config/admins.txt:152]: yeah i forgot to remove it i am so sorry

**Comment by flleeppyy** [comment]
> > Doesn't service cyborg already have a rag built into it?
> 
> image holy shit

okay, so.. I originally worked on this on Monke, not tg, so I instead made a PR here, and I didn't even see that there was a rag added to the service cyborg.

**Comment by flleeppyy** [comment]
my first pr to tgstation and i already fucked up so hard

File: code/game/objects/items/robot/items/hypo.dm [hunk 2]
```diff
@@ -354,6 +378,18 @@
...
+	if(iscyborg(user))
+		var/mob/living/silicon/robot/cyborg = user
+		var/obj/item/borg/apparatus/beaker/service/beverage_apparatus = (locate() in cyborg.model.modules) || (locate() in cyborg.held_items)
+
+		if (isnull(beverage_apparatus))
+			to_chat(user, span_warning("This unit has no beverage apparatus. This shouldn't be possible. Delete yourself, NOW!"))
+			data["apparatusHasItem"] = FALSE
+		else
+			data["apparatusHasItem"] = !isnull(beverage_apparatus.stored)
+	// monkestation end
```
Comment (ZeWaka) [inline code/game/objects/items/robot/items/hypo.dm:392]: but where did it begin

https://www.youtube.com/watch?v=QdCAtXENxBc

File: tgui/packages/tgui/interfaces/BorgShaker.tsx [hunk 3]
```diff
@@ -29,7 +36,60 @@ export const BorgShaker = (props) => {
...
+                  ContainerPreference.InternalBeaker
+                    ? 'green'
+                    : 'default'
+                }
+                tooltip="Search source: Internal Beaker"
+                onClick={() => {
+                  act('set_preferred_container', {
+                    value: ContainerPreference.InternalBeaker,
+                  });
+                  act('update_ui');
```
Comment (ZeWaka) [inline tgui/packages/tgui/interfaces/BorgShaker.tsx:70]: why

File: tgui/packages/tgui/interfaces/BorgShaker.tsx [hunk 4]
```diff
@@ -29,7 +36,60 @@ export const BorgShaker = (props) => {
...
+                  data.reagentSearchContainer ===
+                  ContainerPreference.BeverageApparatus
+                    ? 'green'
+                    : 'default'
+                }
+                onClick={() => {
+                  act('set_preferred_container', {
+                    value: ContainerPreference.BeverageApparatus,
+                  });
+                  act('update_ui');
```
Comment (ZeWaka) [inline tgui/packages/tgui/interfaces/BorgShaker.tsx:87]: why

**Comment by flleeppyy** [comment]
@ZeWaka okay can you, like. be more descriptive and informative? This is my first time touching tgui. my changes work, I don't see what's wrong with them. You commenting "why" does not help me learn at all.

**Comment by flleeppyy** [comment]
also should I make a a `small damp rag` that can't smother people with it's contents, and or be filled?

File: tgui/packages/tgui/interfaces/BorgShaker.tsx (hunk 3 above)
Comment (ZeWaka) [reply tgui/packages/tgui/interfaces/BorgShaker.tsx:70]: rephrased: why are you doing that update ui call

**Comment by flleeppyy** [comment]
> > Sorry for the branch/repo confusion, I'm going to make a separate github account for tgstation itself since github wont let me have tgstation, and a fork of tgstation, on my account lol.
> 
> you can make branches of your fork! image image image name it, make sure tgstation/tgstation is the source, and there you go! image

unfortunately github does not let me select that fork's upstream (aka, monkestation), so.. cant do tgstation

**Comment by excalisword** [comment]
> also should I make a a `small damp rag` that can't smother people with it's contents, and or be filled? just for the service borg.

I don't think I've ever seen the rag used that way by service borgs, so you probably don't need to make a special borg only rag.

File: config/admins.txt (hunk 1 above)
Comment (Sadboysuss) [reply config/admins.txt]: Localhost automatically gives you max perms, so I don't think you even need to do this when testing.

Comment (flleeppyy) [reply config/admins.txt]: I do my development on Linux, so me connecting to my laptop is not localhost (10.0.0.30:1337)

Comment (Sadboysuss) [reply config/admins.txt]: > I do my development on Linux, so me connecting to my laptop is not localhost (10.0.0.30:1337)

Scuffed.

**Pull request merged**